apps/wallet/src/ui/app/redux/slices/sui-objects/index.ts

```diff
@@ -39,12 +41,31 @@ export const fetchAllOwnedAndRequiredObjects = createAsyncThunk<
     void,
     AppThunkConfig
 >('sui-objects/fetch-all', async (_, { getState, extra: { api } }) => {
-    const address = getState().account.address;
+    const state = getState();
+    const {
+        account: { address },
+    } = state;
     const allSuiObjects: SuiObject[] = [];
     if (address) {
         const allObjectRefs =
             await api.instance.fullNode.getObjectsOwnedByAddress(`${address}`);
-        const objectIDs = allObjectRefs.map((anObj) => anObj.objectId);
+        const objectIDs = allObjectRefs
+            .filter((anObj) => {
+                const fetchedVersion = getObjectVersion(anObj);
+                const storedObj = suiObjectsAdapterSelectors.selectById(
+                    state,
+                    getObjectId(anObj)
+                );
+                const storedVersion = storedObj
+                    ? getObjectVersion(storedObj.reference)
+                    : null;
+                const objOutdated = fetchedVersion !== storedVersion;
+                if (!objOutdated && storedObj) {
+                    allSuiObjects.push(storedObj);
+                }
+                return objOutdated;
+            })
+            .map((anObj) => anObj.objectId);
```

Comment:
@pchrysochoidis how do we handle updating the local storage if an object is deleted today? If an object is deleted, it will not show up in `allObjectRefs` and it seems like that the object will never get deleted from the local storage?

Reply:
we assume that all the objects are the ones in the `allSuiObjects` variable. And set this as the new state. (so we clear the current state and updated with this) [see here](https://github.com/MystenLabs/sui/blob/main/apps/wallet/src/ui/app/redux/slices/sui-objects/index.ts#L159)